packages/block-library/src/button/edit.native.js

```diff
@@ -432,19 +486,33 @@ class ButtonEdit extends Component {
 export default compose( [
 	withInstanceId,
 	withColors( 'backgroundColor', { textColor: 'color' } ),
-	withSelect( ( select ) => {
+	withSelect( ( select, { clientId } ) => {
 		const { isEditorSidebarOpened } = select( 'core/edit-post' );
-		const { getSelectedBlockClientId, getBlockParents } = select(
-			'core/block-editor'
-		);
+		const {
+			getSelectedBlockClientId,
+			getBlockCount,
+			getBlockRootClientId,
+		} = select( 'core/block-editor' );
 
+		const parentId = getBlockRootClientId( clientId );
 		const selectedId = getSelectedBlockClientId();
-		const hasParents = getBlockParents( selectedId ).length > 0;
+		const isSelectedButtonsBlock = parentId === selectedId;
```

Comment:
Hmmm, i would rename it to `isButtonsBlockSelected` or to keep the naming convention of block list `isParentSelected`